x/txfees/keeper/txfee_filters/arb_tx.go

```diff
@@ -118,6 +123,26 @@ func isArbTxLooseAuthz(msg sdk.Msg, swapInDenom string, lpTypesSeen map[gammtype
 		return swapInDenom, false
 	}
 
+	// Detect a contract execution via IBC hooks
+	if msgIBCRecv, ok := msg.(*channeltypes.MsgRecvPacket); ok {
```

Comment:
nit: it might be great to if `msgSomething, ok := msg.(type)` grouped under a switch statement
something like
```golang
switch msg.(type):
// ...
case *channeltypes.MsgRecvPacket:
// ...
case poolmanagertypes.SwapMsgRoute:
 return isArbTxLooseSwapMsg(swapMsg, swapInDenom)

return swapInDenom, false
```
Imo, it would make the code more readable